oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/authentication/OAuth2EndpointUtils.java

```diff
@@ -54,6 +57,49 @@ static MultiValueMap<String, String> getParameters(HttpServletRequest request) {
 		return parameters;
 	}
 
+	static MultiValueMap<String, String> getFormParameters(HttpServletRequest request) {
```

Comment:
This method can be simplified. Given the existing `getParameters()` method, the only change required to make it function as `getFormParameters()` is:

```java
...

parameterMap.forEach((key, values) -> {
	if (!request.getQueryString().contains(key) &&	// If not query parameter then it's a form parameter
			values.length > 0) {
		for (String value : values) {
			parameters.add(key, value);
		}
	}
});

...
```

And for `getQueryParameters()`, applying `if (request.getQueryString().contains(key) ...` would work.